bug: C compiler -Winfinite-recursion warning leaks through to user output

## Description

When a Grayscale program contains infinite recursion, clang's `-Winfinite-recursion` warning leaks through to the user **before** the Grayscale runtime's own `P0003` diagnostic fires. Users should never see raw C compiler warnings.

The root cause is that `grayc/src/main.c` invokes the C compiler with `-Wall` but does not include `-Wno-infinite-recursion` to suppress this specific warning. The runtime already guards against this via `gray_enter_func()` and panics with `P0003` at the configured depth limit.

## Minimal Reproduction

```gray
do recurse() {
    recurse()
}

do main() {
    recurse()
}
```

## Actual Output

```
/tmp/gray_gray_run_65419.c:14:35: warning: all paths through this function will call itself [-Winfinite-recursion]
panic[P0003]: maximum recursion depth exceeded (10000 calls deep) at /tmp/gray_gray_run_65419.c:1
```

## Expected Output

```
panic[P0003]: maximum recursion depth exceeded (10000 calls deep) at test.gray:2
```

Only the Grayscale diagnostic should be visible — no raw C compiler warnings.

## Fix

Add `-Wno-infinite-recursion` to the `cc` invocation flags in `grayc/src/main.c` (around line 1700).

--- a/README.md
+++ b/README.md
@@ -1,9 +1,9 @@
 <p align="center">
     <img src="images/Grayscale_logo.png" alt="Grayscale Logo" width="200">
 </p>
-<p align="center">
-  <a href="STANDARD.md">Learn More About Grayscale</a>
+    <h2 align="center">The Grayscale Programming Language</h2>
 </p>
+
 
 <p align="center">
   <a href="https://github.com/grayscale-lang/grayscale/actions/workflows/ci.yml"><img src="https://github.com/grayscale-lang/grayscale/actions/workflows/ci.yml/badge.svg" alt="CI"></a>

--- a/grayc/src/main.c
+++ b/grayc/src/main.c
@@ -1697,7 +1697,7 @@ int main(int argc, char **argv) {
     if (has_archive) {
         snprintf(cmd, sizeof(cmd),
             "cc -std=c11 %s -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable "
-            "-Wno-tautological-compare "
+            "-Wno-tautological-compare -Wno-infinite-recursion "
             "-isystem '%s'/runtime -isystem '%s'/stdlib "
             "-o '%s' '%s' '%s' "
             "-lm -lpthread -Wl,-w 2>&1",
@@ -1736,7 +1736,7 @@ int main(int argc, char **argv) {
 
         snprintf(cmd, sizeof(cmd),
             "cc -std=c11 %s -Wall -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable "
-            "-Wno-tautological-compare "
+            "-Wno-tautological-compare -Wno-infinite-recursion "
             "-isystem '%s'/runtime -isystem '%s'/stdlib "
             "-o '%s' '%s' %s"
             "-lm -lpthread -Wl,-w 2>&1",

